spec/unit/knife/winrm_spec.rb

```diff
@@ -443,6 +443,22 @@
       expect { @winrm.run_with_pretty_exceptions }.to raise_error(SystemExit)
     end
 
+    context "impact of concurrency value" do
```

Comment:
I'd like to see these tests verifying that the requested concurrency limits are respected; or that no concurrency limit is put in place when it's not specified. That is something like: 

``` 
context "impact of concurrency value" do 
  context "when concurrency limit is not set" do 
    it "spawns a number of connection threads equal to the number of target nodes" do 
    end 
  end
  context "when concurrency limit is set" do 
     it "starts only the required number of threads when there are fewer targets than threads" do 
    end  
     it "starts only the requested number of threads when there are as many targets as threads" do 
    end 
     it "starts only the requested number of threads when there are more targets then threads" do 
    end 
   end 
end
```

That first test under when "concurrency limit is set" will fail if implemented as written now, because we start C threads even if number of targets < C. That seems like overhead we could avoid.